app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsModel.kt

```diff
@@ -107,4 +120,76 @@ class AppSettingsModel @Inject constructor(
 
     val pushEndpoint = preference.unifiedPushEndpointFlow()
 
+    private val _pushDistributor = MutableStateFlow<String?>(null)
+    val pushDistributor = _pushDistributor.asStateFlow()
+
+    private val _pushDistributors = MutableStateFlow<List<PushDistributorInfo>?>(null)
+    val pushDistributors = _pushDistributors.asStateFlow()
+
+    /**
+     * Loads the push distributors configuration:
+     *
+     * - Loads the currently selected distributor into [pushDistributor].
+     * - Loads all the available distributors into [pushDistributors].
+     * - If there's only one push distributor available, and none is selected, it's selected automatically.
+     * - Makes sure the app is registered with UnifiedPush if there's already a distributor selected.
+     */
+    private suspend fun loadPushDistributors() {
+        val savedPushDistributor = UnifiedPush.getSavedDistributor(context)
+        _pushDistributor.emit(savedPushDistributor)
+
+        val pushDistributors = UnifiedPush.getDistributors(context)
+            .map { pushDistributor ->
+                try {
+                    val applicationInfo = pm.getApplicationInfo(pushDistributor, 0)
+                    val label = pm.getApplicationLabel(applicationInfo).toString()
+                    val icon = pm.getApplicationIcon(applicationInfo)
+                    PushDistributorInfo(pushDistributor, label, icon)
+                } catch (_: PackageManager.NameNotFoundException) {
+                    // The app is not available for some reason, do not include the app data.
+                    PushDistributorInfo(pushDistributor)
+                }
+            }
+        _pushDistributors.emit(pushDistributors)
+
+        // If there's only one distributor, select it by default.
+        if (pushDistributors.size == 1 && savedPushDistributor == null) {
+            updatePushDistributor(pushDistributors.first().packageName)
+        }
+
+        // If there's already a distributor configured, register the app
+        UnifiedPush.getAckDistributor(context)?.let {
+            UnifiedPush.registerApp(context)
+        }
```

Comment:
I wonder whether we should do this here.

As we now have the dialog, I think we can wait until the user selects an app and clicks OK, which will call the update callback and call `registerApp`.

Reply:
> As we now have the dialog, I think we can wait until the user selects an app and clicks OK, which will call the update callback and call registerApp.

Then, no automatic selection 👍🏼 . In fact, if we ever have automatic selection of the push distributor, it would for sure not be placed on the settings page, so it makes sense to get rid of it, and wait until the user makes a selection